solidity/random-beacon/contracts/libraries/Relay.sol

```diff
@@ -103,13 +109,15 @@ library Relay {
 
         uint64 currentRequestId = ++self.requestCount;
 
-        self.currentRequest = Request(
+        self.currentRequestID = currentRequestId;
+        self.currentRequestGroupID = groupId;
+        self.currentRequestStartBlock = uint64(block.number);
```

Comment:
This will... truncate?

> We did not introduce checks for explicit type conversions from larger to smaller types, because we think that such checks might be unexpected. We would love to get your input on that matter!

https://blog.soliditylang.org/2020/10/28/solidity-0.8.x-preview/

On the other hand:

> Explicit type conversions will always truncate and never cause a failing assertion with the exception of a conversion from an integer to an enum type.

https://docs.soliditylang.org/en/v0.8.10/control-structures.html#checked-or-unchecked-arithmetic

But I think the one from Solidity docs is about `unchecked`.

Reply:
It will truncate, but in practice it would take a few billion years to reach the limit of `uint64` in block numbers so it doesn't matter unless I'm seriously mistaken about something